logic/extensions/atrium-logic-kotlin_1_3/src/commonMain/kotlin/ch/tutteli/atrium/logic/kotlin_1_3/impl/DefaultResultAssertions.kt

```diff
@@ -33,8 +36,42 @@ class DefaultResultAssertions : ResultAssertions {
     override fun <TExpected : Throwable> isFailureOfType(
         container: AssertionContainer<out Result<*>>,
         expectedType: KClass<TExpected>
-    ):  SubjectChangerBuilder.ExecutionStep<Throwable?, TExpected> =
-        container.manualFeature(EXCEPTION) { exceptionOrNull() }.transform().let { previousExpect ->
+    ): SubjectChangerBuilder.ExecutionStep<Throwable?, TExpected> =
+        container.manualFeature(EXCEPTION) {
+
+
+            //fix is here for bug in kotlin 1.3, 1.4, 1.5 (fixed in 1.6) => https://youtrack.jetbrains.com/issue/KT-50974
+            //Comments below explain the fix
+            val badResult = exceptionOrNull()
+            //mapping manually to get internal value throwable - exception or null just produced null + we lost the exception
+            val maybeSubjectResult = container.maybeSubject.map {
+                onSuccess { null }
+                onFailure { it.cause }
+            }
```

Comment:
sorry, I did not really find the time beforehand to take a proper look. This code does nothign as your onSuccess and onFailure only perform side effects. They don't return something which means, maybeSubjectResult is the same as `container.maybeSubject`. You don't have to do anything. I will improve the code and let you know my result in the end

Reply:
Ah I see the error I made sorry about that. Would the solution to be to return some sort of value from the success or failure states and check that?

Reply:
the statement doesn't do anything and you still have fixed the issue which means these lines we can just delete :)